packages/story-editor/src/app/media/utils/useMediaInfo.js

```diff
@@ -0,0 +1,260 @@
+              acc.colorSpace = track.ColorSpace; // Maybe useful in the future.
+            }
+
+            if (track['@type'] === 'Video') {
+              acc.duration = Number(track.Duration);
+              acc.videoCodec = track.Format.toLowerCase();
+            }
+
+            if (track['@type'] === 'Audio') {
+              acc.audioCodec = track.Format.toLowerCase();
+            }
+
+            return acc;
+          },
+          {
+            mimeType: file.type,
+          }
+        );
+
+        normalizedResult.isMuted = !normalizedResult.audioCodec;
+
+        mediaInfo.close();
+
+        return normalizedResult;
+      } catch (err) {
+        // eslint-disable-next-line no-console -- We want to surface this error.
+        console.error(err);
+
+        trackError('mediainfo', err.message);
+
+        return null;
+      } finally {
+        trackTiming();
+      }
+    },
+    [mediainfoUrl]
+  );
+
+  const isConsideredOptimized = useCallback(
+    /**
+     * Determines whether a video file is to be considered optimized.
+     *
+     * Checks things like dimensions, file size, mime type, and codecs.
+     *
+     * @todo Allow WebM with VP9 once Safari catches up.
+     * @param {Resource} resource Resource object.
+     * @param {File} file File object.
+     * @return {Promise<boolean>} Whether the file meets optimization criteria.
+     */
+    async (resource, file) => {
+      // This should never happen, but just in case.
+      if (resource.isOptimized) {
+        return true;
+      }
+
+      // Short-circuit for non-matching mime types.
+      if (!MEDIA_MIME_TYPES_OPTIMIZED_VIDEOS.includes(resource.mimeType)) {
+        return false;
+      }
+
+      // Placeholders are the size of the canvas, so account for that when
+      // checking the dimensions.
+      if (
+        !resource.isPlaceholder &&
+        resource.width * resource.height >
+          MEDIA_VIDEO_DIMENSIONS_THRESHOLD.WIDTH *
+            MEDIA_VIDEO_DIMENSIONS_THRESHOLD.HEIGHT
+      ) {
+        return false;
+      }
+
+      const fileInfo = await getFileInfo(file);
+
+      if (!fileInfo) {
+        return false;
+      }
+
+      // The recommendation for videos in stories is to be < 15s in duration
+      // and < 4 MB in size. If uploading a longer video,
+      // it is only natural that it will exceed the size limit.
+      // Thus, we're instead checking for the average size per second.
+      // Example: 4MB for a 15s video, 12MB for a 45s long video.
+      // TODO: Revisit to avoid fallacy that we're OK with such large file sizes.
+      const hasSmallFileSize =
+        fileInfo.fileSize <
+        (MEDIA_VIDEO_FILE_SIZE_THRESHOLD /
+          MEDIA_RECOMMENDED_MAX_VIDEO_DURATION) *
+          fileInfo.duration;
+
+      const hasSmallDimensions =
+        fileInfo.width * fileInfo.height <=
+        MEDIA_VIDEO_DIMENSIONS_THRESHOLD.WIDTH *
+          MEDIA_VIDEO_DIMENSIONS_THRESHOLD.HEIGHT;
```

Comment:
Isn't this a repeat of the check above?

Reply:
The check above only works for resources that are not placeholders, i.e. it won't work for MKV files for example.

This check here works for MKV files too.